File: app/src/main/java/com/arielfaridja/ezrahi/entities/Latlng.java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by FernFlower decompiler)
//

package com.arielfaridja.ezrahi.entities;

import androidx.annotation.NonNull;

import com.google.firebase.firestore.GeoPoint;

import org.json.JSONException;
import org.json.JSONObject;

public class Latlng {
    private double latitude;
    private double longitude;

    public Latlng(double latitude, double longitude) {
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public Latlng() {
        this.latitude = 0.0D;
        this.longitude = 0.0D;
    }

    public Latlng(Latlng latlng) {
        this.latitude = latlng.getLatitude();
        this.longitude = latlng.getLongitude();
    }

    public Latlng(GeoPoint point) {
        this.latitude = point.getLatitude();
        this.longitude = point.getLongitude();
    }

    public static Latlng StringToLatlng(String string) throws JSONException {
        Latlng res = new Latlng();
        JSONObject j = new JSONObject('{' + string + '}');
        j = new JSONObject(j.get("Latlng").toString());
        res.setLatitude(Double.parseDouble(j.get("latitude").toString()));
        res.setLongitude(Double.parseDouble(j.get("longitude").toString()));
        return res;
    }

    public double getLatitude() {
        return this.latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return this.longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    @NonNull
    @Override
    public String toString() {
        return "Latlng: {latitude=" + this.latitude + ", longitude=" + this.longitude + '}';
    }
}

File: app/src/main/java/com/arielfaridja/ezrahi/entities/Latlng.kt
package com.arielfaridja.ezrahi.entities

import com.google.firebase.firestore.GeoPoint
import org.json.JSONException
import org.json.JSONObject

class Latlng() {
    var latitude: Double = 0.0
    var longitude: Double = 0.0

    constructor(latitude: Double, longitude: Double) : this() {
        this.latitude = latitude
        this.longitude = longitude
    }

    constructor(latlng: Latlng) : this(latlng.latitude, latlng.longitude)
    constructor(point: GeoPoint) : this(point.latitude, point.longitude)

    companion object {
        @JvmStatic
        fun StringToLatlng(string: String): Latlng {
            val res = Latlng()
            val j = JSONObject("{" + string + "}")
            val latlngObj = JSONObject(j.get("Latlng").toString())
            res.latitude = latlngObj.get("latitude").toString().toDouble()
            res.longitude = latlngObj.get("longitude").toString().toDouble()
            return res
        }
    }

    override fun toString(): String {
        return "Latlng: {latitude=$latitude, longitude=$longitude}"
    }
}
